bor/handler.go

```diff
@@ -44,43 +47,29 @@ func HandleMsgProposeSpan(ctx sdk.Context, msg types.MsgProposeSpan, k Keeper) s
 		return common.ErrSpanNotFound(k.Codespace()).Result()
 	}
 
-	// check all conditions
-	if lastSpan.ID+1 != msg.ID || msg.StartBlock < lastSpan.StartBlock || msg.EndBlock < msg.StartBlock {
+	// Validate span continuity
+	if lastSpan.ID+1 != msg.ID || msg.StartBlock != lastSpan.EndBlock+1 || msg.EndBlock < msg.StartBlock {
 		k.Logger(ctx).Error("Blocks not in countinuity",
 			"lastSpanId", lastSpan.ID,
-			"lastSpanStartBlock", lastSpan.StartBlock,
 			"spanId", msg.ID,
+			"lastSpanStartBlock", lastSpan.StartBlock,
+			"lastSpanEndBlock", lastSpan.EndBlock,
 			"spanStartBlock", msg.StartBlock,
+			"spanEndBlock", msg.EndBlock,
 		)
 		return common.ErrSpanNotInCountinuity(k.Codespace()).Result()
 	}
 
-	// freeze for new span
-	err = k.FreezeSet(ctx, msg.ID, msg.StartBlock, msg.ChainID)
-	if err != nil {
-		k.Logger(ctx).Error("Unable to freeze validator set for span", "Error", err)
-		return common.ErrUnableToFreezeValSet(k.Codespace()).Result()
-	}
-
-	// get last span
-	_, err = k.GetLastSpan(ctx)
-	if err != nil {
-		k.Logger(ctx).Error("Unable to fetch last span", "Error", err)
-		return common.ErrSpanNotFound(k.Codespace()).Result()
+	// Validate Span duration
+	spanDuration := k.GetParams(ctx).SpanDuration
+	if spanDuration != (msg.EndBlock - msg.StartBlock) {
```

Comment:
Shouldn't it be as followed:

```suggestion
	if spanDuration != (msg.EndBlock - msg.StartBlock + 1) {
```